Improved

abilogos opened the pull request:
Hello @dilab .

I have made a couple of improvement in order to revive the library with new dependency versions.
fixed test cases depredations.
& a few bugfixes

abilogos committed:
replicate black-bit changes - return status code

abilogos committed:
support for monolog 3 has been added

abilogos committed:
Update composer.json

abilogos committed:
Update composer.json

change the monolog version

abilogos committed:
Update composer.json

abilogos committed:
added instanceId as constructor param in order to isolate uploads and prevent name collission. it can be set to user_id to prevent collission

abilogos committed:
Bugfix: when a file chunk has been uploaded. it didn't report the finished state.

abilogos committed:
fix problem with race condition in directory creation

abilogos committed:
fix bug regarding cakephp TMP not defined on upload pause

abilogos committed:
add new examples to readme

abilogos committed:
bugfix: wrong assumption about the last chunk

abilogos committed:
refactor SimpleResponse

abilogos committed:
update phpunit to version 10

abilogos committed:
make use of ResumableTotalChunks requestParam

williamdes commented:
That's funny, I did quite some similar changes in https://github.com/code-lts/resumable.php

And infact you closed a bug: #30 by 5172968603afe4c60ffaeba89d754c2ed7220056
I fixed the tests this PR broke in #37

abilogos commented:
@williamdes yes that was a wired one since it was considering the chunk before the last one as the final chunk.

but after I think I had fixed that, I have found out it raised another issue.
then after investigating the resumablejs front-end library I have found out that some configuration there to make the the last chunk bigger than the other unless you make `forceChunkSize` option as true.

for example : if your uploading file is 3.5 MB and your chunk size is 1 MB => it chunks your file into 1MB 1MB 1.5MB
and if you set the  `forceChunkSize` option it would chunk the file as 1MB 1MB 1MB 0.5MB

and in the resumable.php the assumption was `forceChunkSize` is always true (which was defaulted to false)
and it was considering the-one-before-last as the final chunk.

but hopefully the resuamblejs sends a header `resumableTotalChunks` so with use of that, we dont need to calculate the last chunk in the backend, and the bug which made the trouble would be fixed.
so I have fixed the new issue with this https://github.com/dilab/resumable.php/pull/36/commits/72f438c03ac14013ceffbda003643e4f0a026dac and the problem get solved

williamdes commented:
Exactly, I came to the same conclusion and implemented total chunks 😄
